internal/management/controller/instance_controller.go

```diff
@@ -394,30 +394,25 @@ func (r *InstanceReconciler) verifyParametersForFollower(cluster *apiv1.Cluster)
 		return err
 	}
 	log.Info("Found previous run flag", "filename", filename)
-	enforcedParams, err := postgresManagement.GetEnforcedParametersThroughPgControldata(r.instance.PgData)
+	controldataParams, err := postgresManagement.LoadEnforcedParametersFromPgControldata(r.instance.PgData)
+	if err != nil {
+		return err
+	}
+	clusterParams, err := postgresManagement.LoadEnforcedParametersFromCluster(cluster)
 	if err != nil {
 		return err
 	}
 
-	clusterParams := cluster.Spec.PostgresConfiguration.Parameters
 	options := make(map[string]string)
-	for key, enforcedparam := range enforcedParams {
+	for key, enforcedparam := range controldataParams {
```

Comment:
Isn't this logic exactly the same as what we have here? https://github.com/cloudnative-pg/cloudnative-pg/pull/4564/files#diff-7b2e81429f27e9e75a48e70527067a4d8a80a2450dab880b4b228d501359b7a8R688

If it is, better encapsulate and use it two times.

Reply:
The one in instance mgr, we only refresh if pgcontrol is higher than user specific, make sure the standby has a higher value.  In restore, we use the higher value between use defined and pgcontrol.

Reply:
I think it is the same, as you're always going to get a value from `controldataParams[X]`, and that value will always be greater than `0`.

Reply:
you are right, I misunderstood

Reply:
Yep, perhaps it was better for me to file this comment against the other occurrence of that cycle.

Reply:
Let's leave them as they are, refactoring it now is way too much pain for the advantage we get.